test/student/StudentController.e2e.ts

```diff
@@ -0,0 +1,59 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { INestApplication } from '@nestjs/common';
+import * as request from 'supertest';
+import { InMemorySchoolRepository } from 'src/school/repository/InMemorySchoolRepository';
+import { SchoolRepository } from 'src/school/repository/SchoolRepository';
+import { School } from 'src/school/model/School';
+import { StudentRepository } from 'src/student/repository/StudentRepository';
+import { InMemoryStudentRepository } from 'src/student/repository/InMemoryStudentRepository';
+import { StudentController } from 'src/student/controller/StudentController';
+import { StudentService } from 'src/student/service/StudentService';
+
+describe('StudentController (e2e)', () => {
+  let app: INestApplication;
+  let studentRepository: StudentRepository;
+  let schoolRepository: SchoolRepository;
+
+  beforeEach(async () => {
+    studentRepository = new InMemoryStudentRepository();
+    schoolRepository = new InMemorySchoolRepository();
+
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      controllers: [StudentController],
+      providers: [
+        {
+          provide: 'StudentRepository',
+          useValue: studentRepository,
+        },
+        {
+          provide: 'SchoolRepository',
+          useValue: schoolRepository,
+        },
+        StudentService,
+      ],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    await app.init();
+  });
+
+  test('/student/new-student (POST)', async () => {
+    const school = new School(1, 9, 'colegio gabo');
+    await schoolRepository.save(school);
+    await request(app.getHttpServer())
+      .post('/student/new-student/')
+      .send({
+        id: 3,
+        schoolId: 1,
+        name: 'sebastian',
+        lastName: 'restrepo',
+        email: 'sebastian@gmail.com',
+        password: '12345',
+      })
+      .expect(201);
```

Comment:
te falto hacer un expect validando la response. Podria ser algo como asi..
```suggestion
.expect(201).
.expect({  
    id: 1,
    schoolId: 1,
    name: 'sebastian',
    lastName: 'restrepo',
    email: 'sebastian@gmail.com'
 });
```